test/index.test.ts

```diff
@@ -64,4 +64,40 @@ describe("Smoke Test", () => {
 
     compareFolders(actualPath, expectedPath);
   });
+
+  test("Should ignore email-test.html", async () => {
+    const { rootDir, actualPath, expectedPath } = await prepare(
+      "with-template-filter",
+    );
+
+    await buildEmailTheme({
+      cwd: rootDir,
+      templatesSrcDirPath: "./fixtures/emails/templates",
+      filterTemplate: (filePath) => !filePath.endsWith(".html"),
+      locales: ["en"],
+      themeNames: ["vanilla"],
+      keycloakifyBuildDirPath: actualPath,
+    });
+
+    compareFolders(actualPath, expectedPath);
+  });
+
+  test("Should break compilation for email-test.html", async () => {
+    const { rootDir, actualPath, expectedPath } = await prepare(
+      "with-template-filter",
+    );
+    try {
+      await buildEmailTheme({
+        cwd: rootDir,
+        templatesSrcDirPath: "./fixtures/emails/templates",
+        locales: ["en"],
+        themeNames: ["vanilla"],
+        keycloakifyBuildDirPath: actualPath,
+      });
+
+      compareFolders(actualPath, expectedPath);
+    } catch (error) {
+      expect(error).toBeTruthy();
+    }
```

Comment:
Something like this would be better
```ts
    await expect(() =>
      buildEmailTheme({
        cwd: rootDir,
        templatesSrcDirPath: "./fixtures/emails/templates",
        locales: ["en"],
        themeNames: ["vanilla"],
        keycloakifyBuildDirPath: actualPath,
      }),
    ).rejects.toThrowError();
```

`compareFolders(actualPath, expectedPath)` is not needed in that case